lib/webengine_kiosk.ex

```diff
@@ -59,19 +52,36 @@ defmodule WebengineKiosk do
   * `width: pixels` - when not fullscreen, the window is this width
   * `height: pixels` - when not fullscreen, the window is this height
   """
+
   @spec start_link(Keyword.t(), GenServer.options()) :: {:ok, pid} | {:error, term}
   def start_link(args, genserver_opts \\ []) do
-    with :ok <- Options.check_args(args) do
-      GenServer.start_link(__MODULE__, args, genserver_opts)
+    # subscriptions are stored in an agent in order to retain them during restarts.
+    {:ok, agent} = Agent.start_link(fn -> [] end)
```

Comment:
This agent should be supervised.

Reply:
Ok i will probably need some advice on this.

This isn't original solution, at least one other library uses this solution (i just forgot where have i stolen this idea). When i was doing it, it seemed to me a bit hacky - and i personally still don't like it but ... How to elegantly solve the problem, that other process needs to tell "here i am and i want so receive events"?

One option is stick the subscribers array directly into WebengineKiosk module - the problem with this is if it crashes, the state is lost and so is the subscribers list.
Second is to not have pubsub, but always send event to the parent process - Circuits libraries does that for example and others as well. Problem with this you can't put the module as Supervisor child easily so user of the library need to take care of this.
Than there is this thing with Agent that relies on fact, that the Agent is way more simple than WebengineKiosk itself and therefore much less likely to crash.

Now there is quick fix for this - put the agent under Supervisor alongside WebengineKiosk and derive its name based on name supplied by the user - this way we can rely on names that doesn't change rather than pids. Still if the Agent crash and is restarted, the subscribers list is lost.

Is there some known agreed way to solve the problem of event registration and dispatch in elixir (without using large third party libraries like event_bus)?

Reply:
Ok  i have an idea how to get this into shape